### PR TITLE
Remove deprecated kubeval and replace with kubeconform

### PR DESCRIPTION
**Pull request by lindluni**
Uses a Docker image instead of downloading binary as an enhancement to https://github.com/github/super-linter/pull/3607

### CODE REVIEW

**Comment by lindluni** [comment]
Missed updating inspec test for proper version flag, will update in the morning.

File: lib/linter.sh [hunk 1]
```diff
@@ -130,7 +130,7 @@ JSCPD_FILE_NAME="${JSCPD_CONFIG_FILE:-.jscpd.json}"
 # shellcheck disable=SC2034  # Variable is referenced indirectly
 JSX_FILE_NAME="${JAVASCRIPT_ES_CONFIG_FILE:-.eslintrc.yml}"
 # shellcheck disable=SC2034  # Variable is referenced indirectly
-KUBERNETES_KUBEVAL_OPTIONS="${KUBERNETES_KUBEVAL_OPTIONS:-null}"
+KUBERNETES_kubeconform_OPTIONS="${KUBERNETES_KUBECONFORM_OPTIONS:-null}"
```
Comment (ferrarimarco) [inline lib/linter.sh:133]: This variable should be all uppercase

File: README.md [hunk 2]
```diff
@@ -51,57 +51,57 @@ The design of the **Super-Linter** is currently to allow linting to occur in **G
```
Comment (ferrarimarco) [inline README.md:51]: We should also update the `VALIDATE_KUBERNETES_KUBEVAL ` and `KUBERNETES_KUBEVAL_OPTIONS` variables for Kubeconform in the README.

File: .automation/test/kubernetes_kubeval/README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
-# Kubeval Test Cases
+# Kubeconform Test Cases
```
Comment (ferrarimarco) [inline .automation/test/kubernetes_kubeval/README.md:1]: Reminder to update the test directory name from `kubernetes_kubeval` to `kubernetes_kubeconform`

**Review by ferrarimarco — CHANGES_REQUESTED** [review]

File: lib/linter.sh (hunk 1 above)
Comment (lindluni) [reply lib/linter.sh:133]: Thank you! Bad find-replace 😂

File: Dockerfile [hunk 4]
```diff
@@ -194,11 +198,11 @@ RUN /install-lintr.sh && rm -rf /install-lintr.sh
...
-###################
-# Install Kubeval #
-###################
-COPY scripts/install-kubeval.sh /
-RUN --mount=type=secret,id=GITHUB_TOKEN /install-kubeval.sh && rm -rf /install-kubeval.sh
+##################
+# Install ktlint #
+##################
+COPY scripts/install-ktlint.sh /
+RUN --mount=type=secret,id=GITHUB_TOKEN /install-ktlint.sh && rm -rf /install-ktlint.sh
```
Comment (lindluni) [inline Dockerfile:205]: This script needs to be broken out into separate scripts on what it actually does in a follow on PR as it's unrelated to this PR

File: lib/functions/linterVersions.sh [hunk 5]
```diff
@@ -75,24 +75,26 @@ BuildLinterVersions() {
...
         # Need specific command for Protolint and editorconfig-checker
         GET_VERSION_CMD="$(echo "--version not supported")"
+      elif [[ ${LINTER} == "checkstyle" ]] || [[ ${LINTER} == "google-java-format" ]]; then
+        GET_VERSION_CMD="$(java -jar "/usr/bin/${LINTER}" --version 2>&1)"
+      elif [[ ${LINTER} == "clippy" ]]; then
+        GET_VERSION_CMD="$(cargo-clippy --version 2>&1)"
+      elif [[ ${LINTER} == "editorconfig-checker" ]]; then
+        GET_VERSION_CMD="$(${LINTER} -version)"
+      elif [[ ${LINTER} == "kubeconform" ]]; then
+        GET_VERSION_CMD="$(${LINTER} -v)"
```
Comment (lindluni) [inline lib/functions/linterVersions.sh:85]: This is just alphabetically arranging the checks

**Comment by lindluni** [comment]
I'd like to cut a release after #3812 is merged before merging this one, as it's a breaking change and will require a major version update

**Review by ferrarimarco — APPROVED** [review]

**Review by ferrarimarco — APPROVED** [review]

**Comment by KingBain** [comment]
For what its worth, the github marketplace is mentioning that super linter is now using kubeconform, having just run github/super-linter@v4.10.1 

It is still running kubeval

not sure when the next release of superlinter will be but their is a mismatch there.

**Pull request merged**